staging/src/k8s.io/cloud-provider/cloud.go

```diff
@@ -98,6 +98,10 @@ func DefaultLoadBalancerName(service *v1.Service) string {
 
 // GetInstanceProviderID builds a ProviderID for a node in a cloud.
 func GetInstanceProviderID(ctx context.Context, cloud Interface, nodeName types.NodeName) (string, error) {
+	if _, ok := cloud.InstancesV2(); ok {
```

Comment:
Would prefer to hoist this check up to getNodeModifiersFromCloudProvider

Reply:
and on line 417, the check should be `if node.Spec.ProviderID == "" && !instancev2Enabled`

Reply:
gotcha, will update!